alerter.js

```diff
@@ -1,26 +1,49 @@
+const { expect } = require("chai");
+const environment = "PRODUCTION";
+
 let alertFailureCount = 0;
+let failureCount = 0;
+/*Setting threshold to 100 deg celcuis*/
+/*if the celcuis value is less than 100 no alert is sent. */
+/*if the celcuis value is greater than 100 alert is sent. */
 
 function networkAlertStub(celcius) {
+  if (celcius < 100) {
+    failureCount += 1;
+    return 500;
+  } else {
     console.log(`Alert! Temperature is ${celcius} degrees`);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
     return 200;
+  }
+}
+
+function networkAlert(celcius) {
+  console.log(`Alert! Temperature is ${celcius} degrees`);
+  return 200;
 }
 
 function alertInCelcius(farenheit) {
-    const celcius = (farenheit - 32) * 5 / 9;
-    const returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
-    }
+  const celcius = ((farenheit - 32) * 5) / 9;
+  const returnCode = null;
+  if (environment === "PRODUCTION") {

```

Comment:
The production code should not be aware of test code or test stub. It should be abstracted by the use of interface or function pointer.